### DISCUSSION TITLE
What is the  format about  index_config and search_config in llama_index？

### DISCUSSION THREAD

**Discussion by Ethereal-sakura** [milvus-io/milvus#31331] [Q&A and General discussion]
There is my code
```python
from llama_index.vector_stores.milvus import MilvusVectorStore
from llama_index.core import VectorStoreIndex, SimpleDirectoryReader
from llama_index.core import StorageContext
documents = SimpleDirectoryReader("./data/paul_graham/").load_data()
print("Document ID:",documents[0].doc_id)
from llama_index.core import StorageContext

index ={
        "M": 32,
        "efConstruction": 100,
        "index_type": "HNSW"  #
}

search_params ={
         "ef":128
}
vector_store = MilvusVectorStore(
    dim=1536,
    overwrite=True,
    uri="http://localhost:19530",
    similarity_metric="l2",
    index_config = index,
    search_config=search_params,
    collection_name="TEST"
)

storage_context = StorageContext.from_defaults(vector_store=vector_store)
index = VectorStoreIndex.from_documents(
    documents, storage_context=storage_context
)
retriever = index.as_retriever()
```
It runs successfully ，but The results displayed above attu indicate that I failed to create the HNSW index,So,What is the right format about index_config and search_config in llama_index？

![image](https://github.com/milvus-io/milvus/assets/110333012/3d9aaa0f-9d68-4ab1-96e9-9a0b57978660)
![image](https://github.com/milvus-io/milvus/assets/110333012/d89b4500-8030-4915-abae-55aa038f6c03)

**Comment by xiaofan-luan** [comment]
1. first of all, even without index you can still search, just the performance will be poor.
2. I didn't see any reason why index build will fail underyour case , we need details logs from your cluster
3. You can always create index on attu or with your own python script and only use llamaindex to insert and search

**Reply by Ethereal-sakura** [reply]
Thanks for your answer ,I suspect that this is a bug on Windows. I have tried many times, and dropped the collection after creation each time, and then re-run this code. However, sometimes attu can show that my index is hnsw, sometimes it is the same as the above screenshot, and sometimes it even shows that the index type is FLAT. This is really confusing to me

**Comment by yhmo** [comment]
I just tested the script with the latest llama_index version. I didn't encounter your problem.
This format is correct:
```
index ={
        "M": 32,
        "efConstruction": 100,
        "index_type": "HNSW"  #
}
```

In the llama_index source code, it pops the "index_type", and the remains "M"/"efConstruction" are passed to "params":
https://github.com/run-llama/llama_index/blob/5c53f41712785e5558156372bdc4f33a6326fa5f/llama-index-integrations/vector_stores/llama-index-vector-stores-milvus/llama_index/vector_stores/milvus/base.py#L334

Finally, the index_config is passed as:
```
index_params = {
    "params": {"M": 32, "efConstruction": 100},
    "metric_type": "L2",
    "index_type": "HNSW",
}
self._collection.create_index(
    self.embedding_field, index_params=index_params
)
```
The logic is no problem.

**Reply by Ethereal-sakura** [reply]
Thanks for your answer ,I have another question,How to use llamaindex to load an exisiting milvus collection？

**Reply by yhmo** [reply]
The MilvusVectorStore only call load() once when index is created: https://github.com/run-llama/llama_index/blob/5c53f41712785e5558156372bdc4f33a6326fa5f/llama-index-integrations/vector_stores/llama-index-vector-stores-milvus/llama_index/vector_stores/milvus/base.py#L343

So, to load an existing collection you don't want to recreate the index, you can use pymilvus to do it.

**Reply by Ethereal-sakura** [reply]
OK,I find the solution about how to load the existing collection,just set the overwrite false

**Reply by yhmo** [reply]
If overwrite=true, the existing collection will be dropped and create new one.

**Comment by Ethereal-sakura** [comment]
May I ask if the index display on attu is correct after you tested my code with this one? Now I have a little doubt that it is the problem of attu. (I am using Windows system) Is there any way to check my index parameters and types through code or command line instead of attu? I am getting anxious about this, looking forward to your reply. Thank you very much



---Original---
From: ***@***.***&gt;
Date: Mon, Mar 18, 2024 12:30 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [milvus-io/milvus] What is the  format about  index_config and search_config in llama_index？ (Discussion #31331)




 
I just tested the script with the latest llama_index version. I didn't encounter your problem.
 This format is correct:
 index ={         "M": 32,         "efConstruction": 100,         "index_type": "HNSW"  # }  
In the llama_index source code, it pops the "index_type", and the remains "M"/"efConstruction" are passed to "params":
 https://github.com/run-llama/llama_index/blob/5c53f41712785e5558156372bdc4f33a6326fa5f/llama-index-integrations/vector_stores/llama-index-vector-stores-milvus/llama_index/vector_stores/milvus/base.py#L334
 
Finally, the index_config is passed as:
 index_params = {     "params": {"M": 32, "efConstruction": 100},     "metric_type": "L2",     "index_type": "HNSW", } self._collection.create_index(     self.embedding_field, index_params=index_params )  
The logic is no problem.

**Reply by yhmo** [reply]
Yes, the index type displayed in Attu is correct. I rerun the script several times, it is always correct.
Milvus: v2.3.10
Attu: v2.3.8
OS: ubuntu 20.04

![Screenshot from 2024-03-18 14-08-36](https://github.com/milvus-io/milvus/assets/2282099/c19a6d5e-1224-4f47-bb5a-dc86d4df7d35)

**Reply by Ethereal-sakura** [reply]
Thank you very much ,so I think it relates to my os system,I will try in ubuntu
